src/pages/AdminLogin.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import { useNavigate } from "react-router-dom";
+import { getAuth, signInWithEmailAndPassword } from "firebase/auth";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Navbar from "../components/Navbar";
+import TextField from "@mui/material/TextField";
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+
+
+function AdminLogin() {
+	const auth = getAuth();
+	const navigate = useNavigate();
+	const [authing, setAuthing] = useState(false);
+	const [user, setUser] = useState({});
+
+	const validationSchema = yup.object({
+		email: yup
+			.string()
+			.email("Enter a valid email")
+			.required("Email is required"),
+		password: yup
+			.string()
+			.required("Password is required")
+	});
+
+
+	const formik = useFormik({
+		initialValues: {
+			email: "",
+			password: ""
+		},
+		validationSchema: validationSchema,
+		onSubmit: (values) => {
+			signInWithEmailAndPassword(
+				auth,
+				values.email,
+				values.password
+			).then((userCredentials) => {
+				setUser(userCredentials.user);
+				setAuthing(true);
+				navigate("/");
+			}).catch((error) => {
+				alert("User not found: " + error.code);
+			});
+		}
+	});
```

Comment:
**thought:** I like using a form helper and validator! I assume more forms will be validated like this later - if it were only used for the login, I'd wonder whether it adds enough value to justify increasing the bundle size with `formik` and `yup` though.

Reply:
We're also going to be using formik and yup for other components, so this will allow us to focus on functionality rather than implementing it from scratch.